civil/civil.go

```diff
@@ -158,6 +159,44 @@ func (d *Date) UnmarshalText(data []byte) error {
 	return err
 }
 
+// Value implements the database/sql/driver Valuer interface.
+func (d Date) Value() (driver.Value, error) {
+	return d.String(), nil
+}
+
+// Scan implements the database/sql Scanner interface.
+func (d *Date) Scan(v any) error {
+	switch vt := v.(type) {
+	case time.Time:
+		*d = DateOf(vt)
+	case *time.Time:
+		if vt != nil {
+			*d = DateOf(*vt)
+		}
+	case string:
+		var err error
+		*d, err = ParseDate(vt)
+		return err
+	case *string:
+		var err error
+		if vt != nil {
+			*d, err = ParseDate(*vt)
+		}
+		return err
+	case []byte:
+		var err error
+		*d, err = ParseDate(string(vt))
+		return err
+	case *[]byte:
+		var err error
+		if vt != nil {
+			*d, err = ParseDate(string(*vt))
+		}
+		return err
+	}
```

Comment:
From: 

```
	// Scan assigns a value from a database driver.
	//
	// The src value will be of one of the following types:
	//
	//    int64
	//    float64
	//    bool
	//    []byte
	//    string
	//    time.Time
	//    nil - for NULL values
	//
	// An error should be returned if the value cannot be stored
	// without loss of information.
```

Two questions:

1. Should int64 values be converted to Date?
2. How should float64 and bool values be handled? Does the error rule above apply here?

Reply:
That's my bad, I should've had a default case that returned an error for other types. There's a case to be made for treating `int64` as a unix timestamp, and we could use some heuristics to guess whether it's seconds, milliseconds, etc., but that feels unnecessary.

Reply:
I don't think so. People store seconds, milliseconds, microseconds, and nanoseconds. If they're doing that, they've already opted out of having date/time conveniences, and I lean towards just erroring on int64, same as we'd error for float64, or bool

Reply:
Hmmm, OK, I suppose we can wait for someone to make a feature request for int64 support if they need it.

Reply:
I added that default error for other scan types, and added additional testing for bad inputs.

Reply:
> The time.Time type accepts a "format" value which may either be a Go identifier for one of the format constants (e.g., "RFC3339") or the format string itself to use with time.Time.Format or time.Parse (https://github.com/golang/go/issues/21990). It can also be "unix", "unixmilli", "unixmicro", or "unixnano" to be represented as a decimal number reporting the number of seconds (or milliseconds, etc.) since the Unix epoch.
https://github.com/golang/go/issues/71497#:~:text=The%20time.Time,the%20Unix%20epoch.

The new json/v2 proposal adds some tag level formatting that could be useful for this in the future. Assuming it gets merged as is, this would be a good route forward, to add some custom json support, and these db functions could use similar tagging to opt into specific formats